File: python/fizzbuzz.py
import sys

def fizzbuzz(max_num):
    """
    Prints the numbers from 1 to max_num, replacing multiples of 3 with "Fizz",
    multiples of 5 with "Buzz", and multiples of both 3 and 5 with "FizzBuzz".

    Args:
        max_num (int): The maximum number to print.

    Returns:
        None
    """
    for num in range(1, max_num + 1):
        if num % 3 == 0 and num % 5 == 0:
            print("FizzBuzz")  # Print "FizzBuzz" if num is divisible by both 3 and 5
        elif num % 3 == 0:
            print("Fizz")  # Print "Fizz" if num is divisible by 3
        elif num % 5 == 0:
            print("Buzz")  # Print "Buzz" if num is divisible by 5
        else:
            print(num)  # Print the number if it is not divisible by 3 or 5

if __name__ == "__main__":
    if len(sys.argv) > 1: # Check if a command line argument was provided
        max_num = int(sys.argv[1]) # Convert the command line argument to an integer
        fizzbuzz(max_num) # Call the fizzbuzz function with the provided argument
    else: # Print an error message if no command line argument was provided
        print("Please provide a maximum number as a command line argument.")

File: c/fizzbuzz.c
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char *argv[]) {
    // Check if the number of command line arguments is correct
    if (argc != 2) {
        printf("Usage: %s <number>\n", argv[0]);
        return 1;
    }

    // Convert the command line argument to an integer
    int number = atoi(argv[1]);

    // Loop through numbers from 1 to the given number
    for (int i = 1; i <= number; i++) {
        // Check if the number is divisible by both 3 and 5
        if (i % 3 == 0 && i % 5 == 0) {
            printf("FizzBuzz\n"); // Print "FizzBuzz" if divisible by both 3 and 5
        } 
        // Check if the number is divisible by 3
        else if (i % 3 == 0) {
            printf("Fizz\n"); // Print "Fizz" if divisible by 3
        } 
        // Check if the number is divisible by 5
        else if (i % 5 == 0) {
            printf("Buzz\n"); // Print "Buzz" if divisible by 5
        } 
        // If none of the above conditions are met, print the number itself
        else {
            printf("%d\n", i);
        }
    }
    return 0;
}